internal/dataplane/sendconfig/dbmode.go

```diff
@@ -67,22 +80,114 @@ func (s UpdateStrategyDBMode) Update(ctx context.Context, targetContent ContentW
 	}
 
 	syncer, err := diff.NewSyncer(diff.SyncerOpts{
-		CurrentState:    cs,
-		TargetState:     ts,
-		KongClient:      s.client,
-		SilenceWarnings: true,
-		IsKonnect:       s.isKonnect,
+		CurrentState:        cs,
+		TargetState:         ts,
+		KongClient:          s.client,
+		SilenceWarnings:     true,
+		IsKonnect:           s.isKonnect,
+		EnableEntityActions: true,
 	})
 	if err != nil {
 		return fmt.Errorf("creating a new syncer for %s: %w", s.client.BaseRootURL(), err), nil, nil
 	}
 
+	ctx, cancel := context.WithCancel(ctx)
+
+	go s.HandleEvents(ctx, syncer.ResultChan)
 	_, errs, _ := syncer.Solve(ctx, s.concurrency, false, false)
+	cancel()
+	resourceErrors = lo.Values(s.resourceErrors)
 	if errs != nil {
-		return deckutils.ErrArray{Errors: errs}, nil, nil
+		return deckutils.ErrArray{Errors: errs}, resourceErrors, nil
+	}
+
+	// as of GDR 1.8 we should always get a plain error set in addition to resourceErrors, so returning resourceErrors
+	// here should not be necessary. Return it anyway as a future-proof because why not.
+	return nil, resourceErrors, nil
+}
+
+// HandleEvents handles logging and error reporting for individual entity change events generated during a sync by
+// looping over an event channel. It terminates when its context dies.
+func (s UpdateStrategyDBMode) HandleEvents(ctx context.Context, events chan diff.EntityAction) {
+	s.resourceErrorLock.Lock()
+	count := 0
+	for {
+		select {
+		case event := <-events:
+			if event.Error == nil {
+				s.logger.V(util.DebugLevel).Info("action", event.Action, "kind", event.Entity.Kind, "name", event.Entity.Name)
+			} else {
+				s.logger.Error(event.Error, "action", event.Action, "kind", event.Entity.Kind, "name", event.Entity.Name)
+				raw, err := resourceErrorFromEntityAction(event)
+				if err != nil {
+					s.logger.Error(err, "could not parse entity update error")
+				} else {
+					rerror, err := parseRawResourceError(raw)
+					if err != nil {
+						s.logger.Error(err, "could not parse entity update error")
+					} else {
+						s.resourceErrors[count] = rerror
```

Comment:
If the index is an incremental `count`, why do you use `map[int]ResourceError` but not `[]ResourceError`?

Reply:
Artifact of some earlier I design I've since forgotten; changed.